test/http2/closing.test.js

```diff
@@ -9,102 +9,94 @@ const { once } = require('node:events')
 const { buildCertificate } = require('../build-certificate')
 const { getServerUrl } = require('../helper')
 
-t.before(buildCertificate)
+test.before(buildCertificate)
 
-t.test('http/2 request while fastify closing', t => {
+test('http/2 request while fastify closing', (t, done) => {
   let fastify
   try {
     fastify = Fastify({
       http2: true
     })
-    t.pass('http2 successfully loaded')
+    t.assert.ok('http2 successfully loaded')
   } catch (e) {
-    t.fail('http2 loading failed', e)
+    t.assert.fail('http2 loading failed')
   }
 
   fastify.get('/', () => Promise.resolve({}))
 
+  t.after(() => { fastify.close() })
   fastify.listen({ port: 0 }, err => {
-    t.error(err)
-    t.teardown(() => { fastify.close() })
-
-    t.test('return 200', t => {
-      const url = getServerUrl(fastify)
-      const session = http2.connect(url, function () {
-        this.request({
-          ':method': 'GET',
-          ':path': '/'
-        }).on('response', headers => {
-          t.equal(headers[':status'], 503)
-          t.end()
-          this.destroy()
-        }).on('error', () => {
-          // Nothing to do here,
-          // we are not interested in this error that might
-          // happen or not
-        })
-        fastify.close()
+    t.assert.ifError(err)
+
+    const url = getServerUrl(fastify)
+    const session = http2.connect(url, function () {
+      this.request({
+        ':method': 'GET',
+        ':path': '/'
+      }).on('response', headers => {
+        t.assert.strictEqual(headers[':status'], 503)
+        done()
+        this.destroy()
+      }).on('error', () => {
+        // Nothing to do here,
+        // we are not interested in this error that might
+        // happen or not
       })
       session.on('error', () => {
         // Nothing to do here,
         // we are not interested in this error that might
         // happen or not
-        t.end()
+        done()
       })
+      fastify.close()
```

Comment:
this should be unneeded with being called in `t.after` [here](https://github.com/fastify/fastify/pull/5757/files#diff-fc24aac3bf47d34df40dc90bc04449121c7fb3e258f2949cc3d0f639b00de29cR27)

Reply:
Let me see... I need to think again about the two first tests.

Reply:
Good call out sorry didnt look at test file